### PR TITLE
Update and expand README.md for grid repo

### PR DESCRIPTION
**Pull request by chenette**
- Update repo description to explain the Hyperledger Grid project and goals

- Summarize the three main grid repos (didn't include grid-website, because it's
  not about the Grid software)

- Add "How to Participate" section with information intended to encourage
  high-level contributions: ideas, use case, and examples

- Remove the "Documentation" section because "how to run and extend Hyperledger
  Grid" distracts from the focus on shaping the project direction

- Add "More Information" section with links

- Clean up the "License" section

Signed-off-by: Anne Chenette <chenette@bitwise.io>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,25 +1,55 @@
-Hyperledger Grid
-----------------
+# Hyperledger Grid
 
-Hyperledger Grid is a project for building supply chain solutions. It includes
-a set of libraries, data models, and SDKs to accelerate development for supply
-chain smart contracts and client interfaces.
+Hyperledger Grid is a framework for building supply chain solutions for
```
Comment (vaporos) [inline README.md]: How about "Hyperleddger Grid is a platform for building supply chain solutions that contain blockchain and distributed ledger components."?

File: README.md [hunk 2]
```diff
@@ -1,25 +1,55 @@
...
-and documented in this [Hyperledger Grid
-Proposal](https://docs.google.com/document/d/1b6ES0bKUK30E2iZizy3vjVEhPn7IvsW5buDo7nFXBE0/edit).
-Information on what _incubation_ means can be found in the [Hyperledger
-Project Lifecycle
-document](https://wiki.hyperledger.org/community/project-lifecycle).
+- The [grid-contrib](https://github.com/hyperledger/grid-contrib) repository
+  contains example middleware tools and reference implementations for smart
+  contracts (also called "transaction families").
+
+- The [grid-rfcs](https://github.com/hyperledger/grid-contrib) repository
```
Comment (peterschwarz) [inline README.md]: Incorrect link.

File: README.md [hunk 3]
```diff
@@ -1,25 +1,55 @@
-Hyperledger Grid
-----------------
+# Hyperledger Grid
 
-Hyperledger Grid is a project for building supply chain solutions. It includes
-a set of libraries, data models, and SDKs to accelerate development for supply
-chain smart contracts and client interfaces.
+Hyperledger Grid is a framework for building supply chain solutions for
+blockchain platforms. It provides a growing set of tools that accelerate
```
Comment (jsmitchell) [inline README.md]: s/blockchain/distributed ledger/

File: README.md [hunk 4]
```diff
@@ -1,25 +1,55 @@
...
+development for supply chain smart contracts and client interfaces.
 
-Documentation
--------------
+This project is not a blockchain platform or distributed ledger application.
+Instead, Hyperledger Grid provides supply-chain-focused libraries, data models,
+and software development kits (SDKs) as modular, reusable middleware between
+existing blockchain platforms and client applications.
```
Comment (jsmitchell) [inline README.md]: s/blockchain/distributed ledger/

File: README.md [hunk 5]
```diff
@@ -1,25 +1,55 @@
...
+  code.
 
-This Hyperledger project is in _incubation_. It was proposed to the community
-and documented in this [Hyperledger Grid
-Proposal](https://docs.google.com/document/d/1b6ES0bKUK30E2iZizy3vjVEhPn7IvsW5buDo7nFXBE0/edit).
-Information on what _incubation_ means can be found in the [Hyperledger
-Project Lifecycle
-document](https://wiki.hyperledger.org/community/project-lifecycle).
+- The [grid-contrib](https://github.com/hyperledger/grid-contrib) repository
+  contains example middleware tools and reference implementations for smart
```
Comment (vaporos) [inline README.md]: Let's get rid of the word "middleware" here.

File: README.md (hunk 1 above)
Comment (chenette) [reply README.md]: Proposed: "Hyperledger Grid is a platform for building supply chain solutions that include distributed ledger components."

File: README.md (hunk 4 above)
Comment (chenette) [reply README.md]: Proposed:

"This project is not an implementation of a distributed ledger or a client       
application. Instead, Hyperledger Grid provides supply-chain-focused libraries, data models, and software development kits (SDKs) as modular, reusable components."

File: README.md (hunk 5 above)
Comment (chenette) [reply README.md]: How about this?

"... example domain models and reference implementations for smart contracts ..."

**Pull request merged**